common/persistence/dataInterfaces.go

```diff
@@ -767,60 +767,29 @@ type (
 
 	// UpdateWorkflowExecutionRequest is used to update a workflow execution
 	UpdateWorkflowExecutionRequest struct {
-		ExecutionInfo    *WorkflowExecutionInfo
-		ReplicationState *ReplicationState
-		VersionHistories *VersionHistories
-		TransferTasks    []Task
-		TimerTasks       []Task
-		ReplicationTasks []Task
-		Condition        int64
-		RangeID          int64
-		ContinueAsNew    *CreateWorkflowExecutionRequest
+		RangeID int64
 
-		// Mutable state
-		UpsertActivityInfos       []*ActivityInfo
-		DeleteActivityInfos       []int64
-		UpserTimerInfos           []*TimerInfo
-		DeleteTimerInfos          []string
-		UpsertChildExecutionInfos []*ChildExecutionInfo
-		DeleteChildExecutionInfo  *int64
-		UpsertRequestCancelInfos  []*RequestCancelInfo
-		DeleteRequestCancelInfo   *int64
-		UpsertSignalInfos         []*SignalInfo
-		DeleteSignalInfo          *int64
-		UpsertSignalRequestedIDs  []string
-		DeleteSignalRequestedID   string
-		NewBufferedEvents         []*workflow.HistoryEvent
-		ClearBufferedEvents       bool
+		UpdateWorkflowMutation WorkflowMutation
+
+		ContinueAsNew *CreateWorkflowExecutionRequest
 
 		Encoding common.EncodingType // optional binary encoding type
 	}
 
 	// ResetMutableStateRequest is used to reset workflow execution state for a single run
 	ResetMutableStateRequest struct {
+		RangeID int64
+
 		// previous workflow information
 		PrevRunID            string
 		PrevLastWriteVersion int64
 		PrevState            int
 
-		ExecutionInfo    *WorkflowExecutionInfo
-		ReplicationState *ReplicationState
-		VersionHistories *VersionHistories
-		Condition        int64
-		RangeID          int64
-
-		// mutable state pending info
-		InsertActivityInfos       []*ActivityInfo
-		InsertTimerInfos          []*TimerInfo
-		InsertChildExecutionInfos []*ChildExecutionInfo
-		InsertRequestCancelInfos  []*RequestCancelInfo
-		InsertSignalInfos         []*SignalInfo
-		InsertSignalRequestedIDs  []string
-
-		// replication/ transfer / timer task
-		InsertReplicationTasks []Task
-		InsertTransferTasks    []Task
-		InsertTimerTasks       []Task
+		// workflow to be resetted
+		ResetWorkflowSnapshot WorkflowSnapshot
```

Comment:
`* WorkflowSnapshot`?

Reply:
no this one is required.
the *WorkflowSnapshot / *WorkflowMutation is optoinal